src/lib/constants.js

```diff
@@ -48,3 +48,19 @@ export const RETURN_KEY = 'Enter';
 export const ESCAPE_KEY = 'Escape';
 export const COMMAND_KEY = 'Meta';
 export const CONTROL_KEY = 'Control';
+
+export const TRACE_TO_AXIS = {
+  cartesian: [
+    'scatter',
+    'box',
+    'bar',
+    'heatmap',
+    'contour',
+    'ohlc',
+    'candlestick',
+  ],
+  ternary: ['scatterternary'],
+  gl3d: ['scatter3d', 'surface', 'mesh3d'],
+  geo: ['scattergeo', 'choropleth'],
+  mapbox: ['scattermapbox'],
+};
```

Comment:
This feels fragile to me - an alternative that wouldn't require any hardcoding here would be to look at `fullTrace._module.basePlotModule.name`. Traces that don't have a subplot, like `pie`, will define their own `basePlotModule`, so then you can filter on names in the list of known subplot types.

Reply:
I'll add this to our list of refactorings in fit n' finish.. thanks..